chalice/deploy/planner.py

```diff
@@ -187,12 +187,17 @@ def _plan_lambdafunction(self, resource):
         # Make mypy happy, it complains if we don't "declare" this upfront.
         params = {}  # type: Dict[str, Any]
         varname = '%s_lambda_arn' % resource.resource_name
+        # Not sure the best way to express this via mypy, but we know
+        # that in the build stage we replace the deployment package
+        # name with the actual filename generated from the pip
+        # packager.  For now we resort to a cast.
+        filename = cast(str, resource.deployment_package.filename)
```

Comment:
Might be worthwhile to add a TODO in the comment so it's easy to find places we might want to periodically check back on when mypy updates and begins to support new things. Though, I suppose just looking for where we cast might also be sufficient.